python/simulation.py

```diff
@@ -3169,3 +3169,52 @@ def quiet(quietval=True):
 
 def verbosity(verbose_val):
     mp.cvar.verbosity = verbose_val
+
+def merge_subgroup_data(data,num_groups):
```

Comment:
Would be nice to compute `num_groups` automatically.   We don't currently store it, but you can compute it via something like:
```py
num_groups = comm.allreduce(int(my_rank() == 0), op=MPI.SUM)
```